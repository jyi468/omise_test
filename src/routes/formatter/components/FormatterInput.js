import React from 'react';

const FormatterInput = (props) => (
    <div className="form-group col-lg-6">
        <label htmlFor="formatterInput">Input (Editable Field)</label>
        <textarea
            className="form-control formatter-input"
            id="formatterInput"
            rows="20"
            onChange={props.onInput}
            placeholder="Input here please..."/>
    </div>
);

export default FormatterInput;