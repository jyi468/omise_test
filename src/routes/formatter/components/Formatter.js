import React, {Component} from 'react';
import FormatterUtils from '../utils/FormatterUtils';
import FormatterInput from './FormatterInput';
import FormatterOutput from './FormatterOutput';

class Formatter extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: [],
            output: []
        };
        this.handleInput = this.handleInput.bind(this);
    }

    handleInput(event) {
        // Format input and update state
        if (event.target) {
            const newOutput = FormatterUtils.buildTree(event.target.value);

            this.setState((prevState, props) =>
                {
                    if (prevState.input !== props.input) {
                        return {
                            output: newOutput
                        };
                    }
                }
            );
        }
    }

    render() {
        // Pass in call back functions to FormatterInput and FormatterOutput
        return (
            <div class="row">
                <div class="col-lg-6">
                    <FormatterInput onInput={this.handleInput}/>
                </div>
                <div class="col-lg-6">
                    <FormatterOutput output={this.state.output}/>
                </div>
            </div>
        );
    }
}

export default Formatter;