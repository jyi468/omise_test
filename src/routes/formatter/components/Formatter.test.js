import React from 'react';
import renderer from 'react-test-renderer';
import Formatter from './Formatter';
import FormatterInput from './FormatterOutput';
import ReactTestUtils from 'react-dom/test-utils';

it('renders output correctly', () => {
    const input = {
        "0":
            [{
                "id": 10,
                "title": "House",
                "level": 0,
                "children": [],
                "parent_id": null
            }],
        "1":
            [{
                "id": 12,
                "title": "Red Roof",
                "level": 1,
                "children": [],
                "parent_id": 10
            }]
    };

    const output = [
        {
            "id": 10,
            "title": "House",
            "level": 0,
            "children": [
                {
                    "id": 12,
                    "title": "Red Roof",
                    "level": 1,
                    "children": [],
                    "parent_id": 10
                }
            ],
            "parent_id": null
        }
    ];

    const tree = ReactTestUtils.renderIntoDocument(<Formatter/>);
    const formatterInput = ReactTestUtils.findRenderedDOMComponentWithClass(tree, 'formatter-input');
    const formatterOutput = ReactTestUtils.findRenderedDOMComponentWithClass(tree, 'formatter-output');

    ReactTestUtils.Simulate.change(formatterInput, {
        target: {
            value: JSON.stringify(input)
        }
    });

    expect(formatterOutput.value).toEqual(JSON.stringify(output, undefined, 2));

    expect(formatterInput).toMatchSnapshot();
    expect(formatterOutput).toMatchSnapshot();
});