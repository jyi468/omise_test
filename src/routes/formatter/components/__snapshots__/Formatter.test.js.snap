// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders output correctly 1`] = `
<textarea
  class="form-control formatter-input"
  id="formatterInput"
  rows="20"
  placeholder="Input here please..."
/>
`;

exports[`renders output correctly 2`] = `
<textarea
  class="form-control formatter-output"
  id="formatterOutput"
  rows="20"
  disabled=""
>
  [ { "id": 10, "title": "House", "level": 0, "children": [ { "id": 12, "title": "Red Roof", "level": 1, "children": [], "parent_id": 10 } ], "parent_id": null } ]
</textarea>
`;
